pkgs/development/python-modules/aider-chat/default.nix

```
@@ -0,0 +1,109 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools
+, pythonOlder
+, universal-ctags
+, aiohttp
+, aiosignal
+, async-timeout
+, attrs
+, backoff
+, certifi
+, charset-normalizer
+, configargparse
+, diskcache
+, frozenlist
+, gitdb
+, gitpython
+, idna
+, jsonschema
+, markdown-it-py
+, mdurl
+, multidict
+, networkx
+, numpy
+, openai
+, packaging
+, prompt-toolkit
+, pygments
+, pyyaml
+, requests
+, rich
+, scipy
+, smmap
+, sounddevice
+, soundfile
+, tiktoken
+, tqdm
+, urllib3
+, wcwidth
+, yarl
+}:
+
+buildPythonPackage rec {
+  pname = "aider-chat";
+  version = "0.14.1";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-8vKvlf9dhJwd9jvHjjVE1JhF3fP9B7U15C42wxct2Qs=";
+  };
+
+  propagatedBuildInputs = [
+    universal-ctags
+    aiohttp
+    aiosignal
+    async-timeout
+    attrs
+    backoff
+    certifi
+    charset-normalizer
+    configargparse
+    diskcache
+    frozenlist
+    gitdb
+    gitpython
+    idna
+    jsonschema
+    markdown-it-py
+    mdurl
+    multidict
+    networkx
+    numpy
+    openai
+    packaging
+    prompt-toolkit
+    pygments
+    pyyaml
+    requests
+    rich
+    scipy
+    smmap
+    sounddevice
+    soundfile
+    tiktoken
+    tqdm
+    urllib3
+    wcwidth
+    yarl
+  ];
+
+  postPatch = ''
+    substituteInPlace aider/repomap.py \
+      --replace '"ctags"' '"${universal-ctags}/bin/ctags"'
+  '';
+
+  # Tests require a Git repository
+  doCheck = false;
+
```

As an additional safe-guard.

```suggestion

  pythonImportsCheck = [
    "aider"
  ];

```